ledger-service/http-json/src/main/scala/com/digitalasset/http/Main.scala

```diff
@@ -50,6 +55,35 @@ object Main {
     config.logLevel.foreach(GlobalLogLevel.set("Ledger HTTP-JSON API"))
   }
 
+  def executeWithShutdownHookNet[F[_], A](
+      program: Deferred[F, Unit] => F[A],
+      shutdownHookTimeout: FiniteDuration = 10.seconds,
+  )(implicit
+      F: Applicative[F],
+      globalConcurrentEffect: ConcurrentEffect[F],
+      globalTimer: Timer[F],
+      loggingCtx: LoggingContextOf[InstanceUUID],
+  ): F[A] = {
+    import cats.FlatMap.ops.toAllFlatMapOps
+    for {
+      _ <- F.pure(()): F[Unit]
+      shutdownHookCalled <- Deferred[F, Unit](globalConcurrentEffect)
+      didProgramAlreadyEnd <- Deferred[F, Unit](globalConcurrentEffect)
```

Comment:
Why is the same effect deferred twice?

Reply:
This is less of something being deferred, it is more about a flag which I use to concurrently check if something is done or not done.